File: api/Entities/Order.cs
namespace AiMenu.Api.Entities;

public class Order
{
    public Guid OrderId { get; set; }
    public Guid RestaurantId { get; set; }
    public Guid TableId { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public string Note { get; set; } = string.Empty;
    public string Status { get; set; } = "Pending";
    public decimal TotalAmount { get; set; }
    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;

    public Restaurant Restaurant { get; set; } = null!;
    public Table Table { get; set; } = null!;
    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}

File: frontend/cashier-web/src/types/order.ts
export type CashierOrderStatus = 'Pending' | 'Preparing' | 'Ready' | 'Paid' | string;

export type CashierOrderListItem = {
  orderId: string;
  restaurantId: string;
  tableId: string;
  tableName: string;
  status: CashierOrderStatus;
  createdAtUtc: string;
  totalAmount: number;
  itemCount: number;
};

export type CashierOrderItem = {
  orderItemId: string;
  productId: string;
  productName: string;
  quantity: number;
  note: string;
  variantName: string;
  unitPrice: number;
  lineTotal: number;
};

export type CashierOrderDetail = {
  orderId: string;
  restaurantId: string;
  tableId: string;
  tableName: string;
  customerName: string;
  note: string;
  status: CashierOrderStatus;
  createdAtUtc: string;
  totalAmount: number;
  items: CashierOrderItem[];
};
